pkg/api/types.go

```diff
@@ -273,15 +273,10 @@ type ExecAction struct {
 	Command []string `json:"command,omitempty"`
 }
 
-// LivenessProbe describes a liveness probe to be examined to the container.
-// TODO: pass structured data to the actions, and document that data here.
-type LivenessProbe struct {
-	// HTTPGetProbe parameters, required if Type == 'http'
-	HTTPGet *HTTPGetAction `json:"httpGet,omitempty"`
-	// TCPSocketProbe parameter, required if Type == 'tcp'
-	TCPSocket *TCPSocketAction `json:"tcpSocket,omitempty"`
-	// ExecProbe parameter, required if Type == 'exec'
-	Exec *ExecAction `json:"exec,omitempty"`
+// Probe describes a liveness probe to be examined to the container.
+type Probe struct {
+	// The action taken to determine the health of a container
+	Handler `json:",inline"`
```

Comment:
I suggested embedding here.  However, I realized this is inconsistent with VolumeSource.

@bgrant0607 preference for style?  The difference in JSON comes down to:

```
"livenessProbe": { "initalDelaySeconds": 10, "exec": { ... } }
"volumes": [ { "name": "vol1", "hostDir": { "path": "/tmp" } } ]
```

vs

```
"livenessProbe": { "initalDelaySeconds": 10, "handler": { "exec": { ... } } }
"volumes": [ { "name": "vol1", "source": { "hostDir": { "path": "/tmp" } } } ]
```